src/ai/backend/agent/agent.py

```diff
@@ -1163,17 +1168,31 @@ async def collect_logs(
         await self.anycast_event(DoSyncKernelLogsEvent(kernel_id, container_id))
 
     async def collect_node_stat(self, interval: float):
+        stat_scope = StatScope.NODE
+        self._stat_task_observer.observe_stat_task_triggered(
+            agent_id=self.id, stat_scope=stat_scope
+        )
         if self.local_config["debug"]["log-stats"]:
             log.debug("collecting node statistics")
         try:
             await self.stat_ctx.collect_node_stat()
+            self._stat_task_observer.observe_stat_task_success(
+                agent_id=self.id, stat_scope=stat_scope
+            )
```

Comment:
Should we work with a decorator? I want the behavior to be guaranteed on success/failure regardless of this function.